src/main/host/descriptor/file.c

```diff
@@ -838,132 +837,313 @@ static inline int _file_getOSDirFD(File* dir) {
     }
 }
 
-int file_fstatat(File* dir, const char* pathname, struct stat* statbuf,
-                 int flags) {
-    trace("File %i fstatat os-backed file %i", dir ? _file_getFD(dir) : 0,
-          _file_getOSDirFD(dir));
+int file_fstatat(File* dir, const char* pathname, struct stat* statbuf, int flags,
+                 const char* workingDir) {
+    int osFd = _file_getOSDirFD(dir);
+    char* pathnameTmp = (char*)pathname;
```

Comment:
Throughout - 

Casting away the const is a bit dangerous, since it could lead to undefined behavior in the future if the underlying storage is const and if we end up mutating the variable. I don't think that currently happens, and *probably* won't happen in the future, but could be a painful bug to track down if it did.

I think we could limit the risk of future UB a bit by declaring these variables `const char*`, and only casting away the `const` in the call to `free`.

Reply:
Changed.